hilti/toolchain/include/ast/node.h

```diff
@@ -484,6 +485,26 @@ class NodeBase : public trait::isNode {
     NodeRef _orig;
 };
 
+
+/** * Common mix-in class for nodes storing doc strings. */
+class NodeWithDocString {
+public:
+    /** Returns the documentation associated with the declaration, if any. */
+    const std::optional<DocString>& documentation() const { return _doc; }
+
+    /** Clears out any documentation associated with the declaration. */
+    void clearDocumentation() { _doc.reset(); }
+
+    /** Sets the documentation associated with the declaration. */
+    void setDocumentation(DocString doc) {
+        if ( doc )
+            _doc = std::move(doc);
```

Comment:
This is slightly confusing: if one passes a falsy `doc`, nothing will get changed. Maybe instead clear the `_doc` in that case instead.

```suggestion
        if ( doc )
            _doc = std::move(doc);
        else
            _doc.reset();
```

Reply:
Ah, yes, I had that at some point, got lost it looks like.